.github/workflows/company4-e2e.yml

```diff
@@ -0,0 +1,105 @@
+name: E2E Testing
+
+on:
+  push: 
+    branches: [main]
+  repository_dispatch:
+    types: 
+      - company4-algorithm-1
+      - company4-algorithm-2
+      - company4-backend
+
+# Make sure URLs defined here correspond with ports defined below
+env:
+  BACKEND_URL: http://localhost:4000
```

Comment:
Any risk of using the same ports here if we run both workflows at once? Not sure if github blocks that or not

Reply:
I don't believe so. I'd add on both your e2e and our e2e workflow the `on:` `pull_request` so they run when we open PRs in this repo. This will test this as well.

Reply:
^ on the PR. Also +1 on no issues with ports, unless we start to mix our images in the same workflow. The workflow runners are isolated from each other so all good there.